Forwarded: https://github.com/ricotz/plank/pull/6
Author: Markus Schmidt <markus262@web.de>, David Mohammed <fossfreedom@ubuntu.com>
Last-Update: 2023-03-19
Description: changed plank positioning according to workarea
 For GNOME based desktops use get_monitor_geometry for screen dimensions

---
 lib/PositionManager.vala | 22 +++++++++++++++++-----
 1 file changed, 17 insertions(+), 5 deletions(-)

diff --git a/lib/PositionManager.vala b/lib/PositionManager.vala
index 0a0928b..ab41259 100644
--- a/lib/PositionManager.vala
+++ b/lib/PositionManager.vala
@@ -65,8 +65,14 @@ namespace Plank
 			screen.composited_changed.connect (screen_composited_changed);
 			
 			// NOTE don't call update_monitor_geo to avoid a double-call of dockwindow.set_size on startup
-			screen.get_monitor_geometry (find_monitor_number (screen, controller.prefs.Monitor), out monitor_geo);
-			
+			var session=Environment.get_variable("XDG_CURRENT_DESKTOP");
+			if (session != null && session.contains("GNOME")) {
+				screen.get_monitor_geometry (find_monitor_number (screen, controller.prefs.Monitor), out monitor_geo);
+			}
+			else {
+				monitor_geo = screen.get_monitor_workarea (find_monitor_number (screen, controller.prefs.Monitor));
+			};
+
 			screen_is_composited = screen.is_composited ();
 		}
 		
@@ -133,9 +139,15 @@ namespace Plank
 		void screen_changed (Gdk.Screen screen)
 		{
 			var old_monitor_geo = monitor_geo;
-			
-			screen.get_monitor_geometry (find_monitor_number (screen, controller.prefs.Monitor), out monitor_geo);
-			
+
+			var session=Environment.get_variable("XDG_CURRENT_DESKTOP");
+			if (session != null && session.contains("GNOME")) {
+				screen.get_monitor_geometry (find_monitor_number (screen, controller.prefs.Monitor), out monitor_geo);
+			}
+			else {
+				monitor_geo = screen.get_monitor_workarea (find_monitor_number (screen, controller.prefs.Monitor));
+			};
+
 			// No need to do anything if nothing has actually changed
 			if (old_monitor_geo.x == monitor_geo.x
 				&& old_monitor_geo.y == monitor_geo.y
-- 
2.34.1

